backend/main.go

```diff
@@ -89,30 +92,77 @@ func (w *Worker) processIncomingEthereumSignatureRequest(sr store.SignatureReque
 	}
 
 	// Ensure incoming tx can still be withdrawn
-	if incomingEthereumTransaction.WithdrawExpiration.After(time.Now()) {
-		return errors.New("transaction withdraw time expired")
+	lastLedgerCloseTime, err := w.Store.GetLastLedgerCloseTime(context.TODO())
+	if err != nil {
+		return errors.Wrap(err, "error getting last ledger close time")
+	}
+
+	if incomingEthereumTransaction.Withdrawn {
+		// TODO on permament errors remove signature request
+		return errors.New("already withdrawn")
+	}
+
+	if lastLedgerCloseTime.After(incomingEthereumTransaction.WithdrawExpiration) {
+		return errors.New("withdrawal no longer possible")
 	}
 
 	outgoingStellarTransaction, err := w.Store.GetOutgoingStellarTransactionForEthereumByHash(context.TODO(), hash)
```

Comment:
it's possible that the `outgoingStellarTransaction` is invalidated because the account sequence of the recipient is bumped. I don't think there's any logic which considers this scenario.

IMO we can remove the state field of outgoingStellarTransaction and we can remove the  outgoingStellarTransaction invalidation logic in the stellar tx observer because anyways we will be calling GetHistoryStellarTransactionByMemoHash().


I think the workflow for processIncomingEthereumSignatureRequest can be:

* check if we are past incomingEthereumTransaction.WithdrawExpiration
* check if w.Store.GetHistoryStellarTransactionByMemoHash() returns a transaction which means the withdrawal was already executed
*  construct and store outgoingStellarTransaction

And for stellar/controllers/stellar_get_inverse_transaction.go the workflow can be:

* check if we are past incomingEthereumTransaction.WithdrawExpiration
* check if w.Store.GetHistoryStellarTransactionByMemoHash() returns a transaction which means the withdrawal was already executed
* if outgoingStellarTransaction exists check that it is not invalidated by looking up account details of the recipient from horizon
* if outgoingStellarTransaction exists and is valid return it in the response, otherwise queue the signature request to the worker